### ISSUE TITLE
Didn't throw warning for the suppression operator

### ISSUE THREAD

**Issue by hez2010** [dotnet/roslyn#45420]
**Version Used**: master 212fc8f

**Steps to Reproduce**:

C# code:
```csharp
var x = 1;
if (!!!(x is not not not not not not 1 !is not false))
{
    Console.WriteLine(x);
}
```

Sharplab: https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQBMQGoA+ABATARgFgAoPABhJLwGYSBvEgAmcb1yZYeJZ8YDcICRgA9GAXkb4A3B16MAlgDNGACgCEGlaPlRGAOwD2MfUZPHD50xcmM1Os40UQANlDgBKd7N5c5c7PgAnFruMtx+AL7eUcQRQA==

**Expected Behavior**:
```
warning IDE0080: Suppression operator has no effect and can be misinterpreted
```
**Actual Behavior**:
No warning

**Comment by CyrusNajmabadi** [comment]
Note: we offer this for type-checks.  Not for general pattern changes:

![image](https://user-images.githubusercontent.com/4564579/85628559-2dffb780-b625-11ea-8256-0c958e1019f2.png)

THe latter case seems much less important.

**Comment by hez2010** [comment]
I meant this one:
```
1 !is not false
  ^
```

**Comment by sharwell** [comment]
@CyrusNajmabadi The case above is nonsensical (no one will write that), but we should verify that IDE0080 is reported for the following:

```
if (x !is true)
{
}
```

**Comment by CyrusNajmabadi** [comment]
We do report that here: 

![image](https://user-images.githubusercontent.com/4564579/222307857-e5ac51e6-72a2-4f4c-b14d-b71666c5b156.png)

Closing out as we do the right thing with reasonable code patterns.
